src/gentropy/dataset/study_index.py

```diff
@@ -183,3 +206,198 @@ def has_summarystats(self: StudyIndex) -> Column:
             Column: True if the study has harmonized summary statistics.
         """
         return self.df.hasSumstats
+
+    def validate_unique_study_id(self: StudyIndex) -> StudyIndex:
+        """Validating the uniqueness of study identifiers and flagging duplicated studies.
+
+        Returns:
+            StudyIndex: with flagged duplicated studies.
+        """
+        validated_df = (
+            self.df.withColumn(
+                "isDuplicated",
+                f.when(
+                    f.count("studyType").over(
+                        Window.partitionBy("studyId").rowsBetween(
```

Comment:
Performance-wise, I think this is a case for a `group by` rather than a window. But I wouldn't worry too much because performance shouldn't be an issue here.

Reply:
The code runs on my laptop, so it is certainly not a bottleneck. I was considering using groupby, but that would be very complicated. Either you have to deal with all the columns, or you do a groupby and join. I think it is fine for now.